crates/router/src/core/payments.rs

```diff
@@ -82,6 +85,11 @@ where
             &merchant_account,
         )
         .await?;
+    authenticate_client_secret(
```

Comment:
This functionality can be modified and added in the Authenticate trait itself.

Reply:
Also could this functionality be in get_trackers itself. So that payment_attempt is not fetched if client_secret auth fails.

Reply:
After offline discussions resolving this conv.